CONTRIBUTING.md

```diff
@@ -179,6 +179,22 @@ $ git commit
 $ git push fork feature
 ```
 
+### Commit Messages
+
+Use descriptive commit messages. Here are [some recommendations](https://cbea.ms/git-commit/)
+on how to write good commit messages.
+When creating PRs GitHub will automatically copy commit messages into the PR description,
+so it is a useful habit to write good commit messages before the PR is created.
+Also, unless you actually want to tell a story with multiple commits make sure to squash
+into a single commit before creating the PR.
```

Comment:
I completely agree.  I don't like reviewing updates to a PR when the latest commit is "all the files" instead of only the files that changed.

We could suggest using `git reset --soft HEAD~X` (where X is the number of commits made locally) before pushing to remote branches.  This allows lots of incremental commits locally while preserving a single useful commit message for the remote.

Reply:
That's exactly what it says:

>make sure to squash into a single commit before creating the PR.

Note _before_.

I don't suggest to squash commits after the PR is created.

Reply:
👍 Makes sense, misread that